crypto/bio/b_sock2.c

```diff
@@ -51,6 +52,17 @@ int BIO_socket(int domain, int socktype, int protocol, int options)
         BIOerr(BIO_F_BIO_SOCKET, BIO_R_UNABLE_TO_CREATE_SOCKET);
         return INVALID_SOCKET;
     }
+# ifndef OPENSSL_NO_KTLS
+    {
+        /*
+         * The new socket is created successfully regardless of ktls_enable.
+         * ktls_enable doesn't change any functionality of the socket, except
+         * changing the setsockopt to enable the processing of ktls_start.
+         * Thus, it is not a problem to call it for non-TLS sockets.
+         */
+        ktls_enable(sock);
```

Comment:
This would be problematic in Linux as the TCP_ULP socket option will fail for sockets that are not in the ESTABLISHED state. I realize that this might be too strict, and we should try to fix it in the Kernel.

Reply:
To be clear, does Linux panic, or does it return an error in this case?  Nothing is checking the error from ktls_enable except in the KTLS tests, and those are only used on non-listen sockets.  If the setsockopt() call fails with an error but otherwise doesn't disturb the socket, then I don't think this will hurt anything?

Reply:
It returns an error without panicking, so it just won't work for BIO_TYPE_CONNECT as it is today. So this change should be fine.

Reply:
This change also enables use of TLS with TOE on Chelsio adapters in Linux.